src/App.tsx

```diff
@@ -1,39 +1,62 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
+type State = {
+  hasClock: boolean;
+  clockName: string,
+};
+
+export class App extends Component<{}, State> {
+  state: State = {
+    hasClock: true,
+    clockName: getRandomName(),
+  };
+
+  timerId = 0;
+
+  componentDidMount() {
+    this.timerId = window.setInterval(() => {
+      const clockName = getRandomName();
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
+      this.setState({ clockName });
+    }, 3300);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('click',
+      this.handleDocumentOnClick, false);
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+    document.addEventListener('contextmenu',
+      this.handleDocumentOnContext, false);
+  }
```

Comment:
U should remove event listeners or use onCliclk and onContexmenu

https://www.geeksforgeeks.org/javascript-removeeventlistener-method-with-examples/